### DISCUSSION TITLE
Can I unit test placement.json

### DISCUSSION THREAD

**Discussion by nogginbox** [OrchardCMS/OrchardCore#9323] [Q&A]
I keep getting placement wrong and end up relying on trial and error to get it right. The cycle time between trial and finding I've errored is quite slow.

I was thinking a unit test would be much quicker to run and also have other benefits.

I'd like to do something like this (warning: made up code):

```
// Arrange
var placement = loadPlacement("mytheme/placement.json");
var shape = Shape.New("TextField");

// Act
placement.Place(shape);

// Assert
Assert.Equal("Footer:1", shape.place);
```

**Comment by Piedone** [comment]
What kind of errors would you like to catch? Because what you described I think can only test what a given shape's placement configuration is (which is apparent from the placement file anyway). It can't tell you if you're using the right shape type/alternate or targeting the right zone. Maybe [Shape Tracing Helpful Extensions](https://github.com/Lombiq/Helpful-Extensions/tree/orchard-core-preview-16154#shape-tracing-helpful-extensions) would be useful for you?
